File: app/src/main/java/org/deiverbum/app/data/entity/LHInvitatoryJoinEntity.java
package org.deiverbum.app.data.entity;

import static org.deiverbum.app.utils.Constants.LH_INVITATORY_JOIN;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */

@Entity(tableName = LH_INVITATORY_JOIN,
        foreignKeys =
        {
                @ForeignKey(
                        entity = LHInvitatoryEntity.class,
                        parentColumns = "caseID",
                        childColumns = "caseFK",
                        onDelete = ForeignKey.CASCADE,
                        onUpdate = ForeignKey.CASCADE),
                @ForeignKey(
                        entity = LHAntiphonEntity.class,
                        parentColumns = "antiphonID",
                        childColumns = "antiphonFK",
                        onDelete = ForeignKey.CASCADE,
                        onUpdate = ForeignKey.CASCADE)
        }
)
public class LHInvitatoryJoinEntity {

    @NonNull
    @PrimaryKey
    @ColumnInfo(name = "groupID")
    public Integer grupoId=0;

    @NonNull
    @ColumnInfo(name = "antiphonFK", index = true)
    public Integer antifonaFK=0;

    @NonNull
    @ColumnInfo(name = "caseFK", index = true)
    public Integer casoFK=0;

    public int getSalmoFK() {
        return  antifonaFK;
    }
    public int getGrupoId() {
        return grupoId;
    }

}

File: app/src/main/java/org/deiverbum/app/data/entity/LHInvitatoryJoinEntity.kt
package org.deiverbum.app.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.Companion.CASCADE
import androidx.room.PrimaryKey
import org.deiverbum.app.util.Constants

/**
 * Entidad para la tabla **`lh_invitatory_join`** de la base de datos, que se ocupa de gestionar la relación asociativa de los Invitatorios de la Liturgia de las Horas.
 *
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@Entity(
    tableName = Constants.LH_INVITATORY_JOIN,
    foreignKeys = [ForeignKey(
        entity = LHInvitatoryEntity::class,
        parentColumns = arrayOf("caseID"),
        childColumns = arrayOf("caseFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LHAntiphonEntity::class,
        parentColumns = arrayOf("antiphonID"),
        childColumns = arrayOf("antiphonFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    )]
)
data class LHInvitatoryJoinEntity(
    @PrimaryKey
    @ColumnInfo(name = "groupID")
    var grupoId: Int,

    @ColumnInfo(name = "antiphonFK", index = true)
    var salmoFK: Int,

    @ColumnInfo(name = "caseFK", index = true)
    var casoFK: Int
)